core/api/api.hpp

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_API_API_HPP
+#define CPP_FILECOIN_CORE_API_API_HPP
+
+#include "crypto/randomness/randomness_types.hpp"
+#include "primitives/block/block.hpp"
+#include "primitives/cid/comm_cid.hpp"
+#include "primitives/rle_bitset/rle_bitset.hpp"
+#include "primitives/ticket/epost_ticket.hpp"
+#include "primitives/ticket/ticket.hpp"
+#include "primitives/tipset/tipset.hpp"
+#include "vm/actor/builtin/market/actor.hpp"
+#include "vm/actor/builtin/payment_channel/payment_channel_actor_state.hpp"
+#include "vm/runtime/runtime_types.hpp"
+
+#define API_METHOD(_name, _result, ...)                                    \
+  struct _##_name : std::function<outcome::result<_result>(__VA_ARGS__)> { \
+    using Result = _result;                                                \
+    using Params = ParamsTuple<__VA_ARGS__>;                               \
+    static constexpr auto name = "Filecoin." #_name;                       \
+  } _name;
+
+namespace fc::api {
+  using common::Buffer;
+  using common::Comm;
+  using crypto::randomness::Randomness;
+  using crypto::signature::Signature;
+  using primitives::ChainEpoch;
+  using primitives::DealId;
+  using primitives::RleBitset;
+  using primitives::SectorNumber;
+  using primitives::SectorSize;
+  using primitives::StoragePower;
+  using primitives::TipsetWeight;
+  using primitives::TokenAmount;
+  using primitives::address::Address;
+  using primitives::block::Block;
+  using primitives::ticket::EPostProof;
+  using primitives::ticket::Ticket;
+  using primitives::tipset::Tipset;
+  using primitives::tipset::TipsetKey;
+  using vm::actor::Actor;
+  using vm::actor::builtin::market::OnChainDeal;
+  using vm::actor::builtin::market::StorageParticipantBalance;
+  using vm::actor::builtin::payment_channel::SignedVoucher;
+  using vm::message::SignedMessage;
+  using vm::message::UnsignedMessage;
+  using vm::runtime::ExecutionResult;
+  using vm::runtime::MessageReceipt;
+
+  template <typename... T>
+  using ParamsTuple =
+      std::tuple<std::remove_const_t<std::remove_reference_t<T>>...>;
+
+  template <typename T>
+  struct Chan {
+    uint64_t id;
+  };
+
+  struct HeadChange {
+    enum class Type { Current, Revert, Apply };
```

Comment:
HeadChange and type already defined [here](https://github.com/filecoin-project/cpp-filecoin/blob/e9938ae7d3c29668ed6e700428d7e4d3194740d6/core/storage/chain/chain_store.hpp#L17)
pls reconcile them.